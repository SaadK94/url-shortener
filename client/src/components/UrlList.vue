<template>
  <UrlItem
    v-for="url in urls"
    :key="url.id"
    :id="url.id"
    :fullUrl="url.fullUrl"
    :shortUrl="url.shortUrl"
    class="item"
    data-jest-item
  />
</template>

<script lang="ts">
import { defineComponent } from "vue";
import UrlItem from "./UrlItem.vue";

export default defineComponent({
  components: {
    UrlItem,
  },
  props: {
    urls: {
      type: Array,
      required: true,
    },
  },
});
</script>

<style scoped>
.item {
  margin-bottom: 16px;
}
</style>
