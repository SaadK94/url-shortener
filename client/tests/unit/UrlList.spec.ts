import { shallowMount, VueWrapper } from "@vue/test-utils";
import UrlList from "@/components/UrlList.vue";

describe("UrlList.vue", () => {
  let wrapper: VueWrapper<any>;

  const props = {
    urls: [
      {
        id: "1",
        fullUrl: "https://www.google.co.uk/",
        shortUrl: "https://pbid.io/abcd1234",
      },
      {
        id: "1",
        fullUrl: "https://primarybid.com/home/",
        shortUrl: "https://pbid.io/abcd9876",
      },
    ],
  };

  beforeEach(() => {
    wrapper = shallowMount(UrlList, { props });
  });

  it("renders the correct number of items", () => {
    expect(wrapper.findAll("[data-jest-item]")).toHaveLength(2);
  });

  it("matches the snapshot", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
