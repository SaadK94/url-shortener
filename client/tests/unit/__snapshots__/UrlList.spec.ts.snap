// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlList.vue matches the snapshot 1`] = `
<div
  data-v-app=""
>
  
  <url-item-stub
    class="item"
    data-jest-item=""
    fullurl="https://www.google.co.uk/"
    id="1"
    shorturl="https://pbid.io/abcd1234"
  />
  <url-item-stub
    class="item"
    data-jest-item=""
    fullurl="https://primarybid.com/home/"
    id="1"
    shorturl="https://pbid.io/abcd9876"
  />
  
</div>
`;
